### PR TITLE
[YUNIKORN-1708] Filtered owner references for placeholder pods.

### PR DESCRIPTION
**Pull request by zhuqi-lucas**
### What is this PR for?
In AWS EMR on EKS service, the driver real pod's ownerReference is configmap.
And placeholder's ownerReference is also the driver configmap.

When user cancels emr-containers job, the job-submitter is terminated,
but the placeholder still remains in pending state.
https://docs.aws.amazon.com/emr/latest/EMR-on-EKS-DevelopmentGuide/emr-eks.html

Make sure configmap owner reference is not returned for pod, it will return the real driver pod itself
because when the placeholder pod set ownerreference to configmap, the configmap will not be deleted for placeholder pod
to GC, we need set the placeholder pod's ownerreference to the real driver pod, when the real driver pod is deleted, the
placeholder will be deleted as well.

The placeholders should really only have one owner: the pod that triggered the creation nothing else.

The placeholder manager should just default to generate just that one.

### What type of PR is it?
* [x] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/YUNIKORN-1708
* Put link here, and add [YUNIKORN-*Jira number*] in PR title, eg. `[YUNIKORN-2] Gang scheduling interface parameters`

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

**Comment by wilfred-s** [comment]
See comments in the jira, different approach is needed

**Comment by zhuqi-lucas** [comment]
Thank you @wilfred-s , i will check it!

File: pkg/appmgmt/general/general.go [hunk 1]
```diff
@@ -104,9 +104,15 @@ func isStateAwareDisabled(pod *v1.Pod) bool {
...
+func getOwnerReference(pod *v1.Pod) []metav1.OwnerReference {
 	if len(pod.OwnerReferences) > 0 {
-		return pod.OwnerReferences
+		// Filter the pod kind owner reference
+		// keep this because we need to find the original pod
+		for _, ref := range pod.OwnerReferences {
+			if ref.Kind == reflect.TypeOf(v1.Pod{}).Name() {
+				return []metav1.OwnerReference{ref}
+			}
+		}
```
Comment (wilfred-s) [inline pkg/appmgmt/general/general.go:115]: This is not needed, the call should only ever be made with the originator or owner pod.
Just always build a new reference without doing anything else.

NIT simplification:
`v1.SchemeGroupVersion.String()` is always "v1" unless we change the signature. 
`reflect.TypeOf(v1.Pod{}).Name()` is always "Pod" nothing else.

File: pkg/appmgmt/general/general_test.go [hunk 2]
```diff
@@ -167,6 +167,7 @@ func TestOriginatorPod(t *testing.T) {
 	owner := apis.OwnerReference{
 		APIVersion: "v1",
 		UID:        "UID-POD-00002",
+		Kind:       "Pod",
 	}
 	refer := []apis.OwnerReference{
 		owner,
```
Comment (wilfred-s) [inline pkg/appmgmt/general/general_test.go:173]: use getOwnerReference to build this

**Review by wilfred-s — CHANGES_REQUESTED** [review]
Two point left and please look at the e2e test failure

File: pkg/appmgmt/general/general.go (hunk 1 above)
Comment (zhuqi-lucas) [reply pkg/appmgmt/general/general.go]: Thank you @wilfred-s for review, addesseed in latest PR!

File: pkg/appmgmt/general/general_test.go [hunk 3]
```diff
@@ -208,10 +210,12 @@ func TestOriginatorPod(t *testing.T) {
 	}
 	am.AddPod(&pod1)
 	assert.Equal(t, len(app.GetNewTasks()), 2)
-	task, err = app.GetTask("UID-POD-00002")
+	task, err = app.GetTask("UID-POD-00001")
```
Comment (zhuqi-lucas) [inline pkg/appmgmt/general/general_test.go:213]: Hi @wilfred-s , if we always return the pod, the test here the originator will be the pod1 who first add into cluster.

**Review by wilfred-s — APPROVED** [review]
LGTM

**Pull request closed without merging**